### PR TITLE
Fix Issue 17630 - selective imports find symbols in private imports o…

### PR DESCRIPTION
**Pull request by RazvanN7**
…f other modules

### CODE REVIEW

**Comment by mathias-lang-sociomantic** [comment]
Sounds like this needs to be a deprecation first :(

**Comment by wilzbach** [comment]
Thanks a lot for moving this forward, but I fear that this needs a deprecation message instead of a hard error.

**Comment by andralex** [comment]
I'm discussing with @RazvanN7, he just pushed a change that issues a deprecation.

**Review by andralex — APPROVED** [review]

File: changelog/fix17630.dd [hunk 1]
```diff
@@ -0,0 +1,5 @@
+Fix Issue 17630 - selective imports find symbols in private imports of other modules
+
+A private module-level import should not have any symbols
+visible outside the module scope. For more information
+see https://issues.dlang.org/show_bug.cgi?id=17630
```
Comment (jacob-carlborg) [inline changelog/fix17630.dd]: I prefer that the changelog entry is extended to explain the issue and the solution inline without having to read the issue.

**Review by jacob-carlborg — CHANGES_REQUESTED** [review]

File: src/dmd/dsymbolsem.d [hunk 2]
```diff
@@ -1145,10 +1145,12 @@ private extern(C++) final class DsymbolSemanticVisitor : Visitor
             sc.protection = imp.protection;
             for (size_t i = 0; i < imp.aliasdecls.dim; i++)
             {
+                bool flag;
                 AliasDeclaration ad = imp.aliasdecls[i];
                 //printf("\tImport %s alias %s = %s, scope = %p\n", toPrettyChars(), aliases[i].toChars(), names[i].toChars(), ad._scope);
-                if (imp.mod.search(imp.loc, imp.names[i]))
+                if (imp.mod.search(imp.loc, imp.names[i]) /*, IgnorePrivateImports*/)
```
Comment (jacob-carlborg) [inline src/dmd/dsymbolsem.d]: Why is this comment here?

**Review by jacob-carlborg — CHANGES_REQUESTED** [review]

File: src/dmd/dsymbolsem.d (hunk 2 above)
Comment (RazvanN7) [reply src/dmd/dsymbolsem.d]: When the deprecation is over, that argument should ne uncommented

Comment (jacob-carlborg) [reply src/dmd/dsymbolsem.d]: Ok.

**Review by jacob-carlborg — APPROVED** [review]

**Comment by andralex** [comment]
Nice work!

**Pull request merged**